### PR TITLE
Add Scheduler thread pool size configuration to configurable variable

### LINKED ISSUE #29729: Move Scheduler thread count to configurable
**Description:**
Currently thread count is provided via `BALLERINA_MAX_POOL_SIZE` move it to a configurable in `lang.runtime`

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
Ballerina runtime provides configuration support for maximum thread pool size. 
It was earlier supported through environment variable `BALLERINA_MAX_POOL_SIZE` 

This PR moves this configuration to configurable variable `STRAND_THREAD_POOL_SIZE` from `lang.runtime` module

Now we can provide the scheduler thread pool size in the `Config.toml` as below,
```toml
[ballerina.lang.runtime]
STRAND_THREAD_POOL_SIZE = 16
```

Fixes #29729 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/constants/RuntimeConstants.java [hunk 1]
```diff
@@ -56,12 +56,13 @@
     public static final String USER_HOME = "user.home";
     public static final String BALLERINA_HOME = "ballerina.home";
     public static final String BALLERINA_PACKAGE_PREFIX = "ballerina" + ORG_NAME_SEPARATOR;
-    public static final String BALLERINA_MAX_POOL_SIZE_ENV_VAR = "BALLERINA_MAX_POOL_SIZE";
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/constants/RuntimeConstants.java:59]: So with this change, we cannot set the pool size with env variable? There are some docs related to that. we might need to change those too.

**Comment by warunalakshitha** [comment]
There are few docs uses env variable to set the pool size. You can find the usages https://github.com/search?p=1&q=org%3Aballerina-platform+BALLERINA_MAX_POOL_SIZE&type=Code

We can omit 1.x version docs. shall we create a doc issue.

**Comment by HindujaB** [comment]
> There are few docs use env variable to set the pool size. You can find the usages https://github.com/search?p=1&q=org%3Aballerina-platform+BALLERINA_MAX_POOL_SIZE&type=Code
> 
> We can omit 1.x version docs. shall we create a doc issue?

Created https://github.com/ballerina-platform/ballerina-dev-website/issues/2147

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java [hunk 2]
```diff
@@ -244,6 +245,22 @@ public void testInvalidDefaultableField() throws BallerinaTestException {
         errorLog.waitForText(5000);
     }
 
+    @Test
+    public void testSchedulerThreadPoolSize() throws BallerinaTestException {
```
Comment (warunalakshitha) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java:249]: I think this is not part of configuration tests. Shall we move to a separate class.

Comment (warunalakshitha) [reply tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java:249]: runtime.SchedulerConfigTest

Comment (warunalakshitha) [reply tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/configurables/ConfigurableTest.java:249]: Should we need 2 tests. 
1 warning and take the env value
2 take from configurable

**Comment by manuranga** [comment]
1) The coding convent for final variable seems to upper case.
2) Shall we make the name bit more descriptive as well, how about `STRAND_THREAD_POOL_SIZE` ?
@sameerajayasoma @warunalakshitha  wtyt?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolTable.java [hunk 3]
```diff
@@ -386,6 +386,7 @@ public void loadPredeclaredModules() {
                                                 Map.entry(Names.INT, this.langIntModuleSymbol),
                                                 Map.entry(Names.MAP, this.langMapModuleSymbol),
                                                 Map.entry(Names.OBJECT, this.langObjectModuleSymbol),
+                                                Map.entry(Names.RUNTIME, this.langRuntimeModuleSymbol),
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolTable.java:389]: This is wrong. Please remove this.

Comment (hasithaa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolTable.java:389]: Ref: https://ballerina.io/ballerina-spec/spec.html#section_8.1

File: langlib/lang.runtime/src/main/ballerina/runtime.bal [hunk 4]
```diff
@@ -18,6 +18,9 @@ import ballerina/jballerina.java;
 import ballerina/lang.'array as lang_array;
 import ballerina/lang.'value as lang_value;
 
+// Maximum thread pool size configuration
+final configurable int STRAND_THREAD_POOL_SIZE = 0;
+
```
Comment (hasithaa) [inline langlib/lang.runtime/src/main/ballerina/runtime.bal:23]: See https://github.com/ballerina-platform/ballerina-spec/issues/838

**Review by hasithaa — CHANGES_REQUESTED** [review]
Need to fixed once we finalized https://github.com/ballerina-platform/ballerina-spec/issues/838

**Pull request closed without merging**